src/main/java/org/opensearch/neuralsearch/processor/AgentStepsResponseProcessor.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.processor;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.extern.log4j.Log4j2;
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.action.search.SearchResponseSections;
+import org.opensearch.neuralsearch.query.ext.AgentStepsSearchExtBuilder;
+import org.opensearch.search.SearchExtBuilder;
+import org.opensearch.search.pipeline.PipelineProcessingContext;
+import org.opensearch.search.pipeline.Processor;
+import org.opensearch.search.pipeline.SearchResponseProcessor;
+
+import java.util.ArrayList;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Response processor that adds agent steps summary to search response extensions
+ */
+@Getter
+@AllArgsConstructor
+@Log4j2
+public class AgentStepsResponseProcessor implements SearchResponseProcessor {
+
+    public static final String TYPE = "agent_steps";
+    @Getter
+    private final String description;
+    @Getter
+    private final String tag;
+    @Getter
+    private final boolean ignoreFailure;
+
+    @Override
+    public SearchResponse processResponse(SearchRequest request, SearchResponse response) {
+        return response;
+    }
+
+    @Override
+    public SearchResponse processResponse(SearchRequest request, SearchResponse response, PipelineProcessingContext requestContext) {
+        if (requestContext == null) {
+            return response;
+        }
+
+        Object agentStepsSummary = requestContext.getAttribute("agent_steps_summary");
+        Object memoryId = requestContext.getAttribute("memory_id");
+
+        if (!(agentStepsSummary instanceof String) || ((String) agentStepsSummary).trim().isEmpty()) {
+            return response;
+        }
+
+        String memoryIdStr = (memoryId instanceof String) ? (String) memoryId : null;
```

Comment:
Shouldn't we throw exception when memoryId is not string type? Also same for the steps summary.
```
if (agentSteps Summary instanceof String == false) {
  throw new RuntimeException("");
}

if (memoryId instanceof String == false) {
  throw new RuntimeException("");
}
```